### ISSUE TITLE
Comments change highlighting of the next line

### ISSUE THREAD

**Issue by pythonpsycho1337** [gpoore/minted#239]
Whenever I add a comment to one of the lines of code i have, it changes the highlighting on the subsequent line. I haven't been able to find any information on how to solve this online. 

Correct highlight: 
![image](https://user-images.githubusercontent.com/6298562/64340880-76db7680-cfe7-11e9-8bf5-3ccd77249965.png)

Incorrect highlighing when using comments(The je and mov instructions are red instead of blue):
![image](https://user-images.githubusercontent.com/6298562/64340788-44ca1480-cfe7-11e9-935e-b22906182028.png)

This is my latex code:
`
\begin{tcolorbox}
\begin{minted}[fontsize=\footnotesize]{gas}
SECTION .data
msg     db      'Hello World!', 0x0A

SECTION .text
start:
        mov edi, 10 
dec:
        cmp edi, 0
        je quit     
print:
        dec edi
        mov edx, 13
        mov ecx, msg
        mov ebx, 1  
        mov eax, 4  
        int 0x80    
        jmp dec    
quit:
        mov ebx, 0  
        mov eax, 1  
        int 0x80    
\end{minted}
\end{tcolorbox}
`
Does any one know a fix for this?

**Comment by gpoore** [comment]
`minted` uses [Pygments](http://pygments.org/) internally to do the highlighting. My guess would be that there is a bug in the lexer. You might take a look at the Pygments documentation for lexers (http://pygments.org/docs/lexerdevelopment/) and the lexer for your language (https://bitbucket.org/birkenfeld/pygments-main/src/default/pygments/lexers/asm.py).

**Comment by mbaz** [comment]
I can't reproduce the error on the console with Pygments version 2.4.2.

**Comment by pythonpsycho1337** [comment]
Solution: I changed the lexer from gas to nasm.
I.e \begin{minted}[fontsize=\footnotesize]{gas} -> \begin{minted}[fontsize=\footnotesize]{nasm}

I am using overleaf.com which, to my understanding, uses tex live 2016. So, it could be possible that it does not use the latest version of pygments. 

I'd say, @mbaz ensure that you are using gas as in the example code i provided. If you still don't get the strange highlighting, the problem is probably solved and we can close this issue.

**Comment by muzimuzhi** [comment]
Compiling the following example on overleaf.com, which is the same as what you provided, the output works fine and with no error.

```latex
\documentclass{article}
\usepackage{minted}
\usepackage{tcolorbox}
\tcbuselibrary{minted}

\begin{document}
\begin{minted}[fontsize=\footnotesize]{gas}
  SECTION .data
  msg db 'Hello World!', 0x0A
  
  SECTION .text
  start:
  mov edi, 10
  dec:
  cmp edi, 0 # comment
  je quit
  print:
  dec edi
  mov edx, 13 ; comment
  mov ecx, msg
  mov ebx, 1
  mov eax, 4
  int 0x80
  jmp dec
  quit:
  mov ebx, 0
  mov eax, 1
  int 0x80
\end{minted}

\begin{tcolorbox}
\begin{minted}[fontsize=\footnotesize]{gas}
  SECTION .data
  msg db 'Hello World!', 0x0A
  
  SECTION .text
  start:
  mov edi, 10
  dec:
  cmp edi, 0 # comment
  je quit
  print:
  dec edi
  mov edx, 13 ; comment
  mov ecx, msg
  mov ebx, 1
  mov eax, 4
  int 0x80
  jmp dec
  quit:
  mov ebx, 0
  mov eax, 1
  int 0x80
\end{minted}
\end{tcolorbox}

\end{document}
```

Hence I cannot reproduce your problem on overleaf too.

**Comment by mbaz** [comment]
@pythonpsycho1337 Yep, I used the gas lexer:

![gas](https://user-images.githubusercontent.com/1630034/64434001-1f084280-d0af-11e9-8fe6-7b67bfe7aad0.png)

This is just a quick test; I also tried with your entire code block.

**Comment by pythonpsycho1337** [comment]
@muzimuzhi I copy pasted your latex code into a new overleaf document and compiled, but the problem still remains for me(See the screenshot below).

This is very strange..

![image](https://user-images.githubusercontent.com/6298562/64558580-78d56a80-d344-11e9-8b55-c70759980b0a.png)

**Comment by pythonpsycho1337** [comment]
Here is a link to the overleaf document: https://www.overleaf.com/7593894324ytvvjhdbkcyp

**Comment by muzimuzhi** [comment]
In short, this is an already-fixed pygments bug. (And hence it has nothing to do with `minted`)

1. Overleaf uses pygments v2.2.0 (Jan 23, 2017)
1. Your problem is fixed by [PR784 of pygments](https://bitbucket.org/birkenfeld/pygments-main/pull-requests/784/fix-handling-of-end-of-line-comments-in/diff), and the oldest version containing this fix is v2.3.1 (Dec 7, 2018)
1. The latest version of pygments is v2.4.2 (May 28, 2019). 

------

This [test.pdf](https://github.com/gpoore/minted/files/3597155/test.pdf), downloaded from your shared overleaf project, provides some evidences. I have added lines to `test.tex` in order to let `test.pdf` carry more info.

`test.pdf` provides two evidences:
1. The wrong highlight is caused by the wrong output of pygments.
You can check this by save the attachment `pygments-output.pygtex` from inside `test.pdf` and check the wrong markup at lines below the line containing comment.
![image](https://user-images.githubusercontent.com/6376638/64639768-1ca14200-d43b-11e9-9f43-d3978aaa6478.png)

2. The first line of `test.pdf` prints the version info of pygments. It shows that overleaf uses pygments v2.2.0, which is released on Jan 23, 2017. You can get full release info [here](https://pypi.org/project/Pygments/#history).

**Comment by muzimuzhi** [comment]
Overleaf is doing some upgrading these days, hence the exact version info may change. But currently I think my last response is the answer to your problem. And, to emphasis again, it shows this problem is about the dependency of `minted`, not `minted` itself.
